pkg/openstack/galera.go

```diff
@@ -28,14 +32,64 @@ const (
 	galeraReady    galeraStatus = iota
 )
 
+func findAndDeleteGalera(
```

Comment:
nit: deleteNotDefinedGaleras() or deleteUndefinedGaleras()
i.e. include that it might delete more than one Galeras and also it deletes those that are not defined any more in the OpenStackControlPlane CR.